bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java

```diff
@@ -246,15 +246,37 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
         Object value = jsonMap.get(key);
         return ObjectUtils.toString(value);
     }
-
+    
     private Relationship createRelationship(Person currentPerson, Map<String, Object> relationshipJson) {
-        Relationship relationship = new Relationship(currentPerson,
-                getPerson((Map<String, Object>) relationshipJson.get("personB")),
-                getRelationshipType((Map<String, Object>) relationshipJson.get("relationshipType")));
+        
+        Relationship relationship;
+        RelationshipType relationshipType = getRelationshipType(
+                (Map<String, Object>) relationshipJson.get("relationshipType"));
+        
+        Person person = getPerson((Map<String, Object>) relationshipJson.get("person"));
+        
+        RelationshipDirection direction = RelationshipDirection.valueOf((String) relationshipJson.get("direction"));
+        if (relationshipJson.get("personB") == null) {
+            direction = RelationshipDirection.A_TO_B;
+        }
+        
+        if (direction.equals(RelationshipDirection.A_TO_B)) {
+            relationship = new Relationship(currentPerson, person,
+                    relationshipType);
+        } else {
+            relationship = new Relationship(person, currentPerson,
+                    relationshipType);
+        }
+        
         relationship.setEndDate(new DateMapper().convertUTCToDate(getValueFromMap(relationshipJson, "endDate")));
-
+        
         return relationship;
     }
+    
+    enum RelationshipDirection {
+        A_TO_B,
+        B_TO_A
+    }
```

Comment:
@Andu033 , could you add that at the very top of the Controller instead? (and make it private)